### DISCUSSION TITLE
Spiral vs Gecko Boot Times

### DISCUSSION THREAD

**Discussion by chaibronz** [SpiralLinux/SpiralLinux-project#376] [Q&A] [answered]
This is prob more of a Debian vs openSUSE question but since the creator of both Spiral and Gecko has expertise in both distros, thought I'd see if I can get some insight here. 

I have installs of both Tumbleweed and Spiral (builder w/ sway) on my laptop. They are both fantastic and to be honest, I'm really impressed with Spiral and likely will be staying here. There's a nice calmness that comes with Debian which I can appreciate. 

The one area Tumbleweed seems to excel in however is boot times, reboot times, shut down times etc  Everything is quick in this department.  With Debian so far, things have been much slower to startup, much slower to reboot, and much slower to shutdown. 

I guess I'm wondering a) if anyone who has run both distros has noticed the same, and b) if any config options can be done on Debian to bring it up to speed with openSUSE? 

And if Sam sees or responds to this, just want to say thanks for all the great work on both projects!

**Comment by AlphaElwedritsch** [comment]
On my system (SpiralLinux with Budgie DM) I have boot- and shutdown times within a few seconds.

**Comment by geckolinux** [comment]
Hi there, thanks to @chaibronz for the kind words. I haven't really noticed any difference between the boot time of the two systems. Pretty much all distros with systemd these days take about the same time to boot, which is usually not notably fast or slow. There's probably some specific service that is hanging during startup on your SpiralLinux installation.

First try to hit `ESC` during bootup to see the startup messages and make note if it hangs or throws a red error for anything.

If nothing stands out there then trying running `sudo systemd-analyze blame` on both GeckoLinux and SpiralLinux.

**Reply by chaibronz** [reply]
Actually boot isn't too bad. It does seem a bit slower with Debian but I also haven't pulled out the stop-watch to actually time the difference lol  Shut down, I am having issues where shut down doesn't work properly with Debian. It seems to shut down quick -- power LED goes off, screen goes off etc  but something is still happening in the background because my battery will be drained next time I go to boot up.  I also noticed that I have to manually hold the power button down for 10-12 seconds after the Debian shut down to make sure it actually shuts down completely.

I'm not sure this is a Debian specific issue though. I also have W10 and openSUSE Aeon installed on the same drive so maybe the three of them aren't getting along. Please consider this a non-issue for now.

**Comment by AlphaElwedritsch** [comment]
> I also noticed that I have to manually hold the power button down for 10-12 seconds after the Debian shut down to make sure it actually shuts down completely.
> 
> I'm not sure this is a Debian specific issue though.

My laptop shuts down straight away. I don't have to hold down the power button.
But I had that too. But it was a BIOS issue. Not a distribution issue.

**Reply by chaibronz** [reply]
How did you fix the BIOS issue? I've been at this for days and no luck. Shuts down fine with openSUSE but not Debian so at least one distribution is handling things fine.

**Reply by AlphaElwedritsch** [reply]
I did a BIOS Update

**Reply by chaibronz** [reply, marked as answer]
Solved!

You got me looking in the right direction. BIOS was up-to-date but there was a setting I had to change.
For any future dummies like myself (Lenovo Thinkpad w/ Intel):

Enter BIOS --> Security --> Security Chip --> Security Chip Selection --> Discrete TPM

I must have changed it to "Intel PTT" at some point, or it's just never come up as an issue before. 

Either way, we're shutting down properly now.
